File: ChatApplicationAPI/ChatApplicationAPI/DAL/Models/Request.cs
using DAL.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAL.Models
{
    public class Request
    {
        public Guid RequestId { get; set; }
        public string SenderId { get; set; }
        public virtual User? Sender { get; set; }
        public string ReceiverId { get; set; }
        public virtual User? Receiver { get; set; }
        public RequestAction Status { get; set; }
        public bool IsTakeAction { get; set; }
        public bool IsDeleted { get; set; }
    }
}

File: ChatApplication/src/app/interfaces/request.ts
import { Guid } from 'guid-typescript';
import { Register } from './register';
import { RequestAction } from '../helpers/RequestAction.enum';

export interface Request {
  requestId: Guid | null;
  senderId: string;
  sender: Register | null;
  receiverId: string;
  receiver: Register | null;
  status: RequestAction;
  isTakeAction: boolean;
}
